app/models/user.rb

```diff
@@ -1,7 +1,11 @@
 class User < ApplicationRecord
   has_one_attached :photo
 
-  has_many :comments
-  has_many :likes
-  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: 'user_id'
+  has_many :likes, foreign_key: 'user_id'
+  has_many :posts, foreign_key: 'author_id'
+
+  def self.recent_3_posts(user)
```

Comment:
- [Optional] Dear @Aqurds, using `self` while creating a method makes the method a `static` method, which means the method `belongs to the class` and not to the `object/user` on which the method is called. Hence every time you call such a method you are expected to pass the user. Instead, a better implementation is to create a nonstatic method/instance method/without `self.method` and use the attribute of the object to perform your operation. 
**This is one of the most powerful weapons of OOP**  
**[Please refer here about static and nonstatic/instance members of the class](https://www.geeksforgeeks.org/ruby-static-members/)**

## Here is a preferable implementation 
```ruby
 def update_post_counter
    author.increment!(:posts_counter)  
  end
```